cmd/common.cpp

```diff
@@ -0,0 +1,309 @@
+    if (cli["--prune.h.older"]->count()) olderHistory.emplace(cli["--prune.h.older"]->as<BlockNum>());
+    if (cli["--prune.r.older"]->count()) olderReceipts.emplace(cli["--prune.r.older"]->as<BlockNum>());
+    if (cli["--prune.s.older"]->count()) olderSenders.emplace(cli["--prune.s.older"]->as<BlockNum>());
+    if (cli["--prune.t.older"]->count()) olderTxIndex.emplace(cli["--prune.t.older"]->as<BlockNum>());
+    if (cli["--prune.c.older"]->count()) olderCallTraces.emplace(cli["--prune.c.older"]->as<BlockNum>());
+
+    db::PruneThreshold beforeHistory, beforeReceipts, beforeSenders, beforeTxIndex, beforeCallTraces;
+    if (cli["--prune.h.before"]->count()) beforeHistory.emplace(cli["--prune.h.before"]->as<BlockNum>());
+    if (cli["--prune.r.before"]->count()) beforeReceipts.emplace(cli["--prune.r.before"]->as<BlockNum>());
+    if (cli["--prune.s.before"]->count()) beforeSenders.emplace(cli["--prune.s.before"]->as<BlockNum>());
+    if (cli["--prune.t.before"]->count()) beforeTxIndex.emplace(cli["--prune.t.before"]->as<BlockNum>());
+    if (cli["--prune.c.before"]->count()) beforeCallTraces.emplace(cli["--prune.c.before"]->as<BlockNum>());
+
+    node_settings.prune_mode =
+        db::parse_prune_mode(prune_mode,  //
+                             olderHistory, olderReceipts, olderSenders, olderTxIndex, olderCallTraces, beforeHistory,
+                             beforeReceipts, beforeSenders, beforeTxIndex, beforeCallTraces);
+
+    // Set chain
+    if (chain_opts_chain_name->count()) {
+        node_settings.network_id = chain_opts_chain_name->as<uint32_t>();
+    }
+}
+
+void run_preflight_checklist(NodeSettings& node_settings) {
+    node_settings.data_directory->deploy();                                  // Ensures all subdirs are present
+    bool chaindata_exclusive{node_settings.chaindata_env_config.exclusive};  // Save setting
+    {
+        auto& config = node_settings.chaindata_env_config;
+        config.path = node_settings.data_directory->chaindata().path().string();
+        config.create =
+            !std::filesystem::exists(db::get_datafile_path(node_settings.data_directory->chaindata().path()));
+        config.exclusive = true;  // Will be cleared after this phase
+    }
+
+    // Open chaindata environment and check tables are consistent
+    log::Message("Opening database", {"path", node_settings.data_directory->chaindata().path().string()});
+    auto chaindata_env{silkworm::db::open_env(node_settings.chaindata_env_config)};
+    db::RWTxn tx(chaindata_env);
+
+    // Ensures all tables are present
+    db::table::check_or_create_chaindata_tables(*tx);
+    log::Message("Database schema", {"version", db::read_schema_version(*tx)->to_string()});
+
+    // Check db is initialized with chain config
+    {
+        node_settings.chain_config = db::read_chain_config(*tx);
+        while (!node_settings.chain_config.has_value()) {
```

Comment:
Do we need `while` here? Perhaps `if` would do?

Reply:
Done